maestrowf/datastructures/core/executiongraph.py

```diff
@@ -392,6 +408,9 @@ def add_step(self, name, step, workspace, restart_limit):
                     "restart_limit": restart_limit,
                 }
         record = _StepRecord(**data)
+        if params:
+            record.add_params(params)
```

Comment:
Should this be a hidden method?

Reply:
Uh, don't think so? That would suggest this sort of external access shouldn't be allowed (i know python doesn't really block that..).

Reply:
Got it, I see what you're saying -- I think I got confused while skimming the changes, but it's a method on the record.